File: sdks/java/src/main/java/com/puxbay/resources/StaffResource.java
package com.puxbay.resources;

import com.puxbay.Puxbay;
import com.puxbay.exceptions.PuxbayException;
import com.puxbay.models.PaginatedResponse;
import com.puxbay.models.Staff;

public class StaffResource {
    private final Puxbay client;

    public StaffResource(Puxbay client) {
        this.client = client;
    }

    public PaginatedResponse<Staff> list(int page, String role) throws PuxbayException {
        String query = "staff/?page=" + page;
        if (role != null) query += "&role=" + role;
        return (PaginatedResponse<Staff>) client.request("GET", query, null, PaginatedResponse.class);
    }

    public Staff get(String staffId) throws PuxbayException {
        return client.request("GET", "staff/" + staffId + "/", null, Staff.class);
    }

    public Staff create(Staff staff) throws PuxbayException {
        return client.request("POST", "staff/", staff, Staff.class);
    }

    public Staff update(String staffId, Staff staff) throws PuxbayException {
        return client.request("PATCH", "staff/" + staffId + "/", staff, Staff.class);
    }

    public void delete(String staffId) throws PuxbayException {
        client.request("DELETE", "staff/" + staffId + "/", null, Void.class);
    }
}

File: sdks/kotlin/src/main/kotlin/com/puxbay/resources/StaffResource.kt
package com.puxbay.resources

import com.puxbay.Puxbay
import com.puxbay.models.Staff
import com.puxbay.models.PaginatedResponse
import io.ktor.http.HttpMethod

class StaffResource(client: Puxbay) : BaseResource(client) {
    suspend fun list(page: Int = 1, role: String? = null): PaginatedResponse<Staff> {
        val roleParam = if (role != null) "&role=$role" else ""
        return client.request(HttpMethod.Get, "staff/?page=$page$roleParam")
    }
    
    suspend fun get(staffId: String): Staff =
        client.request(HttpMethod.Get, "staff/$staffId/")
    
    suspend fun create(staff: Staff): Staff =
        client.request(HttpMethod.Post, "staff/", staff)
    
    suspend fun update(staffId: String, staff: Staff): Staff =
        client.request(HttpMethod.Patch, "staff/$staffId/", staff)
    
    suspend fun delete(staffId: String) {
        client.request<Unit>(HttpMethod.Delete, "staff/$staffId/")
    }
}
